StringVariable.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * StringVariable.h
+ *
+ *  Created on: Jun 20, 2018
+ *      Author: baiyuhuang
+ */
+
+#ifndef STRINGVARIABLE_H_
+#define STRINGVARIABLE_H_
+#include "StringVariable.h"
+#include "MessageConstants.h"
+#include <cstdlib>
+#include "stdio.h"
+#include "string.h"
+#include <typeinfo>
+
+const int DATA_STRING_SIZE = 11;
+
+class StringVariable {
```

Comment:
Define this class as template class.
`template <typename T>`
`class StringVariable {...}`
Then you can remove `template <typename T>` on member function declaration.